packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -19,41 +19,57 @@ import * as path from 'path';
 import * as RequireInTheMiddle from 'require-in-the-middle';
 import { satisfies } from 'semver';
 import { InstrumentationAbstract } from '../../instrumentation';
-import { InstrumentationModuleDefinition } from './types';
+import {
+  InstrumentationBaseNode,
+  InstrumentationModuleDefinition
+} from './types';
 import { diag } from '@opentelemetry/api';
 
 /**
  * Base abstract class for instrumenting node plugins
  */
 export abstract class InstrumentationBase<T = any>
   extends InstrumentationAbstract
-  implements types.Instrumentation {
-  private _modules: InstrumentationModuleDefinition<T>[];
+  implements InstrumentationBaseNode<T> {
+  private _modules?: InstrumentationModuleDefinition<any>[];
   private _hooks: RequireInTheMiddle.Hooked[] = [];
   private _enabled = false;
+  private _timer?: NodeJS.Timeout;
 
   constructor(
     instrumentationName: string,
     instrumentationVersion: string,
     config: types.InstrumentationConfig = {}
   ) {
     super(instrumentationName, instrumentationVersion, config);
+    this._timer = setTimeout(() => {
```

Comment:
This will result in a crash. Is this intended?

Reply:
see here https://github.com/open-telemetry/opentelemetry-js/pull/2417#issuecomment-901440570